From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 15 Mar 2022 15:19:41 +0100
Subject: Use GetComputerName instead of gethostname

(cherry picked from commit 82165d95a925a775f2f895518dba8ff6f1a732d1)
(cherry picked from commit c6bd48f9265da174863861a1289aba5ef7d8eb7e)
---
 channels/rdpdr/client/rdpdr_main.c | 6 +++++-
 libfreerdp/core/settings.c         | 6 ++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/channels/rdpdr/client/rdpdr_main.c b/channels/rdpdr/client/rdpdr_main.c
index 309d24c..681ebce 100644
--- a/channels/rdpdr/client/rdpdr_main.c
+++ b/channels/rdpdr/client/rdpdr_main.c
@@ -31,6 +31,7 @@
 #include <string.h>
 
 #include <winpr/crt.h>
+#include <winpr/sysinfo.h>
 #include <winpr/stream.h>
 
 #include <winpr/sspicli.h>
@@ -1114,7 +1115,10 @@ static UINT rdpdr_send_client_name_request(rdpdrPlugin* rdpdr)
 	size_t computerNameLenW;
 
 	if (!rdpdr->computerName[0])
-		gethostname(rdpdr->computerName, sizeof(rdpdr->computerName) - 1);
+	{
+		DWORD size = sizeof(rdpdr->computerName) - 1;
+		GetComputerNameA(rdpdr->computerName, &size);
+	}
 
 	computerNameLenW = ConvertToUnicode(CP_UTF8, 0, rdpdr->computerName, -1, &computerNameW, 0) * 2;
 	s = Stream_New(NULL, 16 + computerNameLenW + 2);
diff --git a/libfreerdp/core/settings.c b/libfreerdp/core/settings.c
index d0c2b56..bd259d6 100644
--- a/libfreerdp/core/settings.c
+++ b/libfreerdp/core/settings.c
@@ -405,8 +405,10 @@ rdpSettings* freerdp_settings_new(DWORD flags)
 
 	if (!settings->ClientHostname)
 		goto out_fail;
-
-	gethostname(settings->ClientHostname, 31);
+	DWORD size = 31;
+	GetComputerNameA(ClientHostname, &size);
+	if (!freerdp_settings_set_string(settings, FreeRDP_ClientHostname, ClientHostname))
+		goto out_fail;
 	settings->ClientHostname[31] = 0;
 	settings->ColorPointerFlag = TRUE;
 	settings->LargePointerFlag = (LARGE_POINTER_FLAG_96x96 | LARGE_POINTER_FLAG_384x384);
